tests/base/symmetric_tensor_39.cc

```diff
@@ -44,6 +44,9 @@ int main()
           counter += 1.0;
         }
 
+  // without this we get wrong results with clang in release mode
+  std::cout << '\0';
```

Comment:
This does not really print anything. I first tried to print the result after each line in `internal::perform_double_contraction` and noticed then that each call to `cout` solved the issue no matter if in that function or directly before the function call.